scripts/init.mk

```diff
@@ -1,26 +1,32 @@
 # This file is part of the repository template project. Please, DO NOT edit this file.
 
```

Comment:
Could do something like:

```make
install-asdf-dependency:
    ifeq ("${dependency}", "")
        $(error Variable dependency has not been defined)
    endif
         
    asdf plugin add ${dependency}
    asdf install ${dependency}

python-install:
    $(MAKE) install-asdf-dependency dependency=python
    $(MAKE) install-asdf-dependency dependency=poetry
```

And then it could be expanded easier by a developer later down the line either in this file. These commands could be tweaked to read in the `.tool-versions` and automatically add the plugins and runtimes defined in there. Or not - an argument can be made that these commands are simple enough to edit over time.

Reply:
Good stuff. Done! A few comments:

- `_install-dependency` starts with underscore marking it with an intent to be used only internally, therefore defensive coding not needed at this occasion
-   It's recommended to used just `make` instead of `$(MAKE)` and ensure our CLI tooling is setup correctly. Also, in our use case we will never use additional flags like -j for parallel jobs. This also encourages simplicity and clarity
-  Preference is to use bash-like syntax or compatible with an exception if it would make it more difficult to understand, e.g. `$(or ${version},)` is make-native `or`